configure

```diff
@@ -83,6 +83,61 @@ cxx_works()
     fi
 }
 
+# Test if the compiler successfully compiles the specified source file. If the optional compilation
+# argument is specified ($3) then pass it to the compiler along with the already-established
+# arguments (this can be used to test if the compiler accepts the argument).
+#   $1 - the name of the source file to compile
+#   $2 - the name of the result object file
+#   $3 - the argument to test (optional)
+#   CXX - the name of the compiler
+#   CXXFLAGS, CXXFLAGS_EXTRA - any predetermined flags
+try_compile_source()
+{
+    eval $CXX $CXXFLAGS ${3:-} $CXXFLAGS_EXTRA '"$configtmpdir"/"$1"' \
+            -c -o '"$configtmpdir"/"$2"' > /dev/null 2>&1
+}
+
+# Test if the linker successfully links the specified object file. If the optional link argument
```

Comment:
_Test if the specified object file can be linked into an executable_.

This doesn't invoke the linker directly, you're invoking the compiler driver, which perhaps does invoke the linker if it accepts the arguments it's given. It's not just testing the linker.